### ISSUE TITLE
Add support for xsd("auto")

### ISSUE THREAD

**Issue by laurianvm** [vliz-be-opsci/py-sema#116]
consider making the | xsd  filter a bit smarter:
Add xsd("auto"), which applies the best fitting xsd type to the actual datatype

(copying from IMIS-1598)

**Comment by cedricdcc** [comment]
@laurianvm  are you working on this actively?
Also can you explain the diff/example why this issue and issue #115 ?

**Comment by marc-portier** [comment]
@cedricdcc the description seems clear
1. #116 --> `|  xsd("auto")`  will apply an xsd type that matches the python-type passed in (which assumes some smartness in the Source implementation that does proper type conversion, or assumes the template is doing that type conversion)
2. #115 --> `| xsd("detect")  will ignore the input type, assume str (or even force conversion) and then make a guess of the type before handing it over to the appropriate matching xsd-type 

In other words:
`| xsd(detect)` == `| do-str-to-type-detection-and-conversion` +`| xsd(auto)`
